File: java/example/src/main/java/org/example/math/FunctionDefinition.java
package org.example.math;

import java.util.function.Function;

public class FunctionDefinition {
    public static double fDefinition(double number) {
        final Function<Double, Double> f = (Double x) -> 3 * Math.pow(x, 2) + 2 * x - 5;
        return f.apply(number);
    }
}

File: example/src/org/example/math/function_definition.py
from org.example.math.config.logging_config import logger


def f_definition(number):
    """
    :return: f(x) = 3x² + 2x -5
    """

    def f(x):
        return 3 * pow(x, 2) + 2 * x - 5

    result = f(number)
    logger.info("f({0}) = {1}".format(number, result))
    return result
